docs/proposed/http-storage-node-protocol.rst

```diff
@@ -0,0 +1,506 @@
+The storage node will present an x509 certificate during the TLS handshake.
+Storage clients will require that the certificate have a valid signature.
+The Subject Public Key Information (SPKI) hash of the certificate will constitute the storage node's identity.
+The **tub id** portion of the storage node fURL will be replaced with the SPKI hash.
+
+When connecting to a storage node,
+the client will take the following steps to gain confidence it has reached the intended peer:
+
+* It will perform the usual cryptographic verification of the certificate presented by the storage server.
+  That is,
+  it will check that the certificate itself is well-formed,
+  that it is currently valid [#]_,
+  and that the signature it carries is valid.
+* It will compare the SPKI hash of the certificate to the expected value.
+  The specifics of the comparison are the same as for the comparison specified by `RFC 7469`_ with "sha256" [#]_.
+
+To further clarify, consider this example.
+Alice operates a storage node.
+Alice generates a key pair and secures it properly.
+Alice generates a self-signed storage node certificate with the key pair.
+Alice's storage node announces (to an introducer) a fURL containing (among other information) the SPKI hash.
+Imagine the SPKI hash is ``i5xb...``.
+This results in a fURL of ``pb://i5xb...@example.com:443/g3m5...#v=2`` [#]_.
+Bob creates a client node pointed at the same introducer.
+Bob's client node receives the announcement from Alice's storage node.
+
+Bob's client node recognizes the fURL as referring to an HTTP-dialect server due to the ``v=2`` fragment.
+Bob's client node can now perform a TLS handshake with a server at the address in the fURL location hints
+(``example.com:443`` in this example).
+Following the above described validation procedures,
+Bob's client node can determine whether it has reached Alice's storage node or not.
+If and only if the validation procedure is successful does Bob's client node conclude it has reached Alice's storage node.
+**Peer authentication** has been achieved.
+
+Additionally,
+by continuing to interact using TLS,
+Bob's client and Alice's storage node are assured of both **message authentication** and **message confidentiality**.
+
+.. note::
+
+   Foolscap TubIDs are 20 bytes (SHA1 digest of the certificate).
+   They are presented with base32 encoding at a length of 32 bytes.
+   SPKI information discussed here is 32 bytes (SHA256 digest).
+   They will present in base32 as 52 bytes.
+   https://tools.ietf.org/html/rfc7515#appendix-C may prove a better (more compact) choice for encoding the information into a fURL.
+   It will encode 32 bytes into merely 43...
+   We could also choose to reduce the hash size of the SPKI information through use of another cryptographic hash (replacing sha256).
+   A 224 bit hash function (SHA3-224, for example) might be suitable -
+   improving the encoded length to 38 bytes.
+   Or we could stick with the Foolscap digest function - SHA1.
+
```

Comment:
Seems bad to stick with SHA1 (especially if its just so "a chunk of indecipherable text is somewhat shorter")